### PR TITLE
feat: allow reusing external messages in generated handler

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
You can now do

```rust
#[spaad::handler(msg = "HandleEvent")]
pub async fn handle_event(&mut self, msg: HandleEvent) {
    unimplemented!();
}
```

feel free to squash merge

### CODE REVIEW

File: spaad_internal/src/entangle/transform.rs [hunk 1]
```diff
@@ -2,33 +2,92 @@ use crate::entangle::{get_actor_name, get_name, ty_is_name};
...
+    let value = match value {
+        Ok(p) => p,
+        _ => abort!(lit, "value should be a valid type"),
+    };
+
+    value
 }
 
-fn is_context_ref(ty_ref: &TypeReference, is_handler: bool) -> bool {
+fn is_context_ref(ty_ref: &TypeReference) -> bool {
```
Comment (ForsakenHarmony) [inline spaad_internal/src/entangle/transform.rs:83]: `is_handler` was always true

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+root = true
```
Comment (Restioson) [inline .editorconfig:1]: ?

File: spaad_internal/src/entangle/transform.rs [hunk 3]
```diff
@@ -2,33 +2,92 @@ use crate::entangle::{get_actor_name, get_name, ty_is_name};
...
-    attrs.iter().position(|attr| is_attr(name, attr))
+        let (path, nested) = match meta {
+            // #[spaad::thing]
+            Meta::Path(path) => (path, None),
+            // #[spaad::thing(with = args)]
+            Meta::List(list) => (list.path, Some(list.nested)),
+            Meta::NameValue(_) => abort!(meta, "spaad attributes can only be lists (`#[spaad::handler(...)]`) or paths (`#[spaad::handler]`)"),
+        };
+
+        if path.segments.first().unwrap().ident != "spaad" {
```
Comment (Restioson) [inline spaad_internal/src/entangle/transform.rs:31]: Ahhh this is a good way of doing this. Thanks!

**Comment by Restioson** [comment]
Lgtm except I'm not sure what .editorconfig is

**Comment by ForsakenHarmony** [comment]
https://editorconfig.org/

it's just a small file which is meant to work across editors/ides to tell them some basic rules about how to format files

File: .editorconfig (hunk 2 above)
Comment (Restioson) [reply .editorconfig:1]: No I mean what's the file for?

**Comment by Restioson** [comment]
Ah ok fair enough.

**Pull request merged**